### PR TITLE
Thank You! message occasionally when users rate ads

### PR DESCRIPTION
**Pull request by jeremybmerrill**
To "reward" users who rate ads political/normal with a li'l dopamine burst, we display a little "thank you!" message in the popup. The message starts out fairly frequent, but its frequency declines exponentially. 

In English, we choose between 5 playful Thank You! messages; in other languages, it's just that language's basic "Thanks!", in some cases with a smiley face.

Adds a reducer to keep track of the total number of rated ads.

### CODE REVIEW

File: extension/src/popup.jsx [hunk 1]
```diff
@@ -64,13 +65,18 @@ const updateRating = (id, rating) => ({
   id: id,
   value: rating
 });
+const sayThanks = ratings_length => ({
+  type: SAY_THANKS,
+  ratings_length
```
Comment (thejefflarson) [inline extension/src/popup.jsx]: It is slightly confusing to use `ratings_length` here when we've used `ratings_count` everywhere else.

File: extension/src/popup.jsx [hunk 2]
```diff
@@ -114,6 +120,37 @@ const terms = (state = false, action) => {
...
+  "thanks5" //  "Way to go! You've categorized $count$ ads so far."
+];
+const thanks = (state = null, action) => {
+  switch (action.type) {
+    case SAY_THANKS:
+      // we thank you frequently at first and exponentially less often later
+      // for instance, your first ad, we have a 50% chance of thanking you
+      // and your 4th ad a 25% chance and so on.
+      let message =
+        Math.random() < 1 / (2 * Math.sqrt(action.ratings_length || 1))
```
Comment (thejefflarson) [inline extension/src/popup.jsx]: so much fun!

Comment (jeremybmerrill) [reply extension/src/popup.jsx]: I wish I still had my TI-82 to try out the various decay functions; I guess you can never go home again.

File: extension/src/popup.jsx (hunk 1 above)
Comment (jeremybmerrill) [reply extension/src/popup.jsx]: fixed in 591236d

**Pull request merged**